api/preprints/views.py

```diff
@@ -69,6 +69,30 @@
 from osf.metrics import PreprintDownload, PreprintView
 
 
+class RejectPreprintEditMixin:
```

Comment:
I am thinking if we should avoid confusion between rejected preprint by naming this class with a descriptive name like `PreprintOldVersionsImmutableMixin`.